cudax/include/cuda/experimental/__launch/launch_transform.cuh

```diff
@@ -37,7 +36,7 @@ namespace cuda::experimental::detail
 // argument to make this overload less preferred than other overloads that take
 // a stream_ref as the first argument.
 template <typename _Arg>
-_Arg&& __cudax_launch_transform(__ignore, _Arg&& __arg) noexcept
+_CCCL_NODISCARD constexpr _Arg&& __cudax_launch_transform(__ignore, _Arg&& __arg) noexcept
```

Comment:
I believe we rather want to hook up into `get_stream` rather than having the user to manually specify a a function with difficult semantics. 
From a user perspective that would be much easier to do

Reply:
This is not an API for users. It is for us. And I don't know how `get_stream` could replace what this API will do for us.

Reply:
> // Types should define overloads of __cudax_launch_transform that are find-able
> // by ADL in order to customize how cudax::launch handles that type

This sound like you want it to customizable by users and we definitely want the user to have a way of communicating "This object has a stream ordering requirement, this is the stream"

Reply:
We will have various storage types (`vector`) and annotations (`in`, `out`, etc) to control data movement. This CPO is what we will use to tell `launch` how to handle each. It is for injecting before/after actions for each argument, and doing it in a way that is easily extensible. But the extensibility is for us, as the library matures, not for the end user.